api/db.py

```diff
@@ -93,6 +93,9 @@ async def update(self, obj):
         if obj.id is None:
             raise ValueError("Cannot update object with no id")
         col = self._get_collection(obj.__class__)
+        node = await col.find_one({'_id': ObjectId(obj.id)})
```

Comment:
The variable should not be `node` as we don't know what type of object is being updated here. We could find out using `obj.__class__` but it's better to not rely on that in the `Database` class.

Reply:
Yes, I need to rename the `node` obj. Another way to check the object class is `type` operator.

Reply:
We can rely on `obj.__class__` to find the matching database collection in the dictionary using `_get_collection()`, but then we shouldn't be making the logic of the code depend on the class of the model.  Instead, all the models should have the same abstract methods to handle the data.